: Type '({ context, id, innerProps }: ContextModalProps<{    contextProp: string;}>) => JSX.Element' is not assignable to type 'FC<ContextModalProps<{}>>'.

### What package has an issue

@mantine/modals

### Describe the bug

demo code

```typescript
 <ModalsProvider
                modals={{
                  hello: ({
                    context,
                    id,
                    innerProps
                  }: ContextModalProps<{ contextProp: string }>) => {
                    console.log(innerProps)
                    return (
                      <div>
                        <div>Test custom modal: {innerProps.contextProp}</div>
                        <Button onClick={() => context.closeModal(id)}>
                          Close
                        </Button>
                      </div>
                    )
                  }
                }}
                labels={{ confirm: 'Confirm', cancel: 'Cancel' }}
              >
```
```shell
src/layout/wrapper.tsx:44:19 - error TS2322: Type '({ context, id, innerProps }: ContextModalProps<{    contextProp: string;}>) => JSX.Element' is not assignable to type 'FC<ContextModalProps<{}>>'.
  Types of parameters '__0' and 'props' are incompatible.
    Type 'PropsWithChildren<ContextModalProps<{}>>' is not assignable to type 'ContextModalProps<{ contextProp: string; }>'.
      Types of property 'innerProps' are incompatible.
        Property 'contextProp' is missing in type '{}' but required in type '{ contextProp: string; }'.

44                   hello: ({
```

### In which browser did the problem occur

chrome

### If possible, please include a link to a codesandbox with the reproduced problem

_No response_

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

### Possible fix

...

--- a/src/mantine-modals/src/ModalsProvider.tsx
+++ b/src/mantine-modals/src/ModalsProvider.tsx
@@ -18,7 +18,7 @@ export interface ModalsProviderProps {
   children: React.ReactNode;
 
   /** Predefined modals */
-  modals?: Record<string, React.FC<ContextModalProps>>;
+  modals?: Record<string, React.FC<ContextModalProps<any>>>;
 
   /** Shared Modal component props, applied for every modal */
   modalProps?: ModalSettings;

--- a/src/mantine-modals/src/use-modals/use-modals.test.tsx
+++ b/src/mantine-modals/src/use-modals/use-modals.test.tsx
@@ -1,4 +1,4 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, PropsWithChildren } from 'react';
 import ReactDOM from 'react-dom';
 import { renderHook, WrapperComponent } from '@testing-library/react-hooks';
 import { MantineProvider } from '@mantine/core';
@@ -29,7 +29,7 @@ describe('@mantine/modals/use-modals', () => {
   });
 
   it('returns context value of ModalsProvider', () => {
-    const wrapper = ({ children }) => (
+    const wrapper = ({ children }: PropsWithChildren<unknown>) => (
       <MantineProvider>
         <ModalsProvider>{children}</ModalsProvider>
       </MantineProvider>

